### PR TITLE
Upgrade to Rails 7.1

### PR DESCRIPTION
**Pull request by rahearn**
Changes:

* Upgrade to Rails 7.1.x
* Run app:update
* Revert one setting to 7.0 in the test environment to fix a request spec relying on that behavior
* Update rails-ujs integration
* Update to rails 7.1 default config
* Update ruby version to 3.3.4
* Update node version to 22.3.0 and remove some CI cruft
* Bonus scope creep: fix 500 error when filtering a blank search by agency

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -33,34 +33,17 @@ jobs:
       - name: Set up Ruby
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 3.2.2
```
Comment (rahearn) [inline .github/workflows/test.yml:36]: by default, setup-ruby will grab the version from `.ruby-version`

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -33,34 +33,17 @@ jobs:
...
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
-
-      - name: Install linux dependencies
-        run: |
-          sudo apt-get -yqq install libpq-dev build-essential libssl-dev
+          node-version: 22.3.0
+          cache: 'yarn'
```
Comment (rahearn) [inline .github/workflows/test.yml:43]: `cache: 'yarn'` will load and save the cache, but doesn't actually install the dependencies, for some reason

File: .ruby-version [hunk 3]
```diff
@@ -1 +1 @@
-ruby-3.2.4
+3.3.4
```
Comment (mogul) [inline .ruby-version:1]: Latest supported by the ruy-buildpack as of writing... Excellent!

**Review by mogul — APPROVED** [review]
Great stuff! Thanks for paying attention to the available Ruby and Node versions that are in the deployed buildpacks.

**Pull request merged**